pandas/core/generic.py

```diff
@@ -7715,34 +7715,107 @@ def last(self, offset):
     def rank(self, axis=0, method='average', numeric_only=None,
              na_option='keep', ascending=True, pct=False):
         """
-        Compute numerical data ranks (1 through n) along axis. Equal values are
-        assigned a rank that is the average of the ranks of those values
+        Compute numerical data ranks (1 through n) along axis.
+
+        By default, equal values are assigned a rank that is the average of the
+        ranks of those values.
 
         Parameters
         ----------
         axis : {0 or 'index', 1 or 'columns'}, default 0
-            index to direct ranking
-        method : {'average', 'min', 'max', 'first', 'dense'}
-            * average: average rank of group
-            * min: lowest rank in group
-            * max: highest rank in group
-            * first: ranks assigned in order they appear in the array
-            * dense: like 'min', but rank always increases by 1 between groups
-        numeric_only : boolean, default None
+            Index to direct ranking.
+        method : {'average', 'min', 'max', 'first', 'dense'}, default 'average'
+            Which method to use to rank equal values:
+            * average: average rank of group.
+            * min: lowest rank in group.
+            * max: highest rank in group.
+            * first: ranks assigned in order they appear in the array.
+            * dense: like 'min', but rank always increases by 1 between groups.
+        numeric_only : bool, default None
```

Comment:
Probably `optional` would be better than `default None`. What does `None` mean here? Can you explain in the description.

Reply:
Hi @datapythonista,

I am not entirely sure how to document this, since as WillAyd suggested above the functionality may be deprecated (#19560) in future releases.  

Do you think this could be enough?
```
numeric_only : bool, optional
    If True, rank only float, int and boolean data.
    Valid only for DataFrame or Panel objects.
```

Reply:
`For DataFrame objects, rank only numeric columns if set to True.`. for example. Panel is deprecated and almost being removed, so no need to mention.

Reply:
Hi @datapythonista,
Thank you for this.

I think I have adjusted everything, I hope that it is correct now.

Let me know if there is anything else that does not look right.